buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/NamingConventionsTask.groovy

```diff
@@ -78,9 +68,19 @@ public class NamingConventionsTask extends LoggedExec {
         FileCollection extraClasspath = project.configurations.namingConventions
         dependsOn(extraClasspath)
 
+        FileCollection classpath = project.sourceSets.test.runtimeClasspath
+        inputs.files(classpath)
         description = "Runs NamingConventionsCheck on ${classpath}"
+        if (project.sourceSets.names.contains('main')) {
+            /* If there are main classes we'll have a look at them just to make
+             * sure there aren't file named like test classes in there too. */
+            inputs.files(project.sourceSets.main.runtimeClasspath)
```

Comment:
I don't think this should be necessary? The test classpath should already include the main classpath.